resolver/src/resolver_handle/mod.rs

```diff
@@ -259,54 +285,25 @@ impl ResolverFuture {
     ///
     /// # Arguments
     /// * `host` - string hostname, if this is an invalid hostname, an error will be returned.
-    pub fn lookup_ip<N: IntoName + TryParseIp>(&self, host: N) -> LookupIpFuture {
-        let mut finally_ip_addr = None;
-
-        // if host is a ip address, return directly.
-        if let Some(ip_addr) = host.try_parse_ip() {
-            // if ndots are greater than 4, then we can't assume the name is an IpAddr
-            //   this accepts IPv6 as well, b/c IPv6 can take the form: 2001:db8::198.51.100.35
-            //   but `:` is not a valid DNS character, so techinically this will fail parsing.
-            //   TODO: should we always do search before returning this?
-            if self.options.ndots > 4 {
-                finally_ip_addr = Some(ip_addr);
-            } else {
-                return LookupIpFuture::ok(
-                    self.client_cache.clone(),
-                    Lookup::new_with_max_ttl(Arc::new(vec![ip_addr])),
-                );
-            }
-        }
-
-        let name = match (host.into_name(), finally_ip_addr.as_ref()) {
-            (Ok(name), _) => name,
-            (Err(_), Some(ip_addr)) => {
-                // it was a valid IP, return that...
-                return LookupIpFuture::ok(
-                    self.client_cache.clone(),
-                    Lookup::new_with_max_ttl(Arc::new(vec![ip_addr.clone()])),
-                );
-            }
-            (Err(err), None) => {
-                return LookupIpFuture::error(self.client_cache.clone(), err);
-            }
+    pub fn lookup_ip<N: IntoName + TryParseIp>(&self, host: N) -> BackgroundLookupIp {
```

Comment:
Oh, this doesn't have a 'static restriction like I thought, I'm not clear on why the callers (see resolve in the example) has that if it's the case this does not...